storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -95,13 +108,86 @@ func (s *commitLogSource) Read(
 	defer iter.Close()
 
 	var (
-		unmerged    = make(map[uint32]map[ts.Hash]encoderMap)
+		numShards = len(shardsTimeRanges)
+		unmerged  = make([]map[ts.Hash]encoderMap, numShards, numShards)
```

Comment:
Add types for the different sub-parts of `unmerged, and make the operations used in the read/merge phase functions on those types. 

E.g. for the read side, it could be something like:

```
type shardEncoders map[ts.Hash]encoderMap
type shardSetEncoders map[uint32]shardEncoders
func (s *shardSetEncoders) GetOrAdd(shardID uint32) shardEncoders {} 
func (s *shardEncoders) GetOrAdd(seriesID ts.ID) encoderMap {} 
```